doc/source/rllib/core-concepts.rst

```diff
@@ -163,176 +163,162 @@ These batches are wrapped up together in a ``MultiAgentBatch``,
-
-Understanding and Debugging Execution Plans
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Execution plans are based on Ray `parallel iterators <https://github.com/ray-project/ray/blob/master/python/ray/util/iter.py>`__ and can be inspected similarly. For example, suppose you wanted to print out the intermediate data items during training. This can be done by inserting a print function into the dataflow, e.g., for A2C:
-
-.. code-block:: python
+        return train_results
 
-    def debug_print(item):
-        print("I saw", type(item))
-        return item
+As you can see each call to ``training_iteration`` does one sampling and training update. A dictionary is a returned that contains the results of the training update.
+ Its generally recommended that the dictionary map keys of type ``str`` to values that are of type ``float`` or to dictionaries of the same form, allowing for a nested structure.
 
-    train_op = rollouts \
-        .combine(ConcatBatches(
-            min_batch_size=config["train_batch_size"])) \
-        .for_each(debug_print) \
-        .for_each(TrainOneStep(workers))
-
-You'll see output like this on the console:
-
-.. code-block:: bash
+Training Iteration Function Utilities
+~~~~~~~~~~~~~~~~~~~~~~~
 
-    (pid=6555) I saw <class 'ray.rllib.policy.sample_batch.SampleBatch'>
-    (pid=6555) I saw <class 'ray.rllib.policy.sample_batch.SampleBatch'>
-    (pid=6555) I saw <class 'ray.rllib.policy.sample_batch.SampleBatch'>
-    (pid=6555) I saw <class 'ray.rllib.policy.sample_batch.SampleBatch'>
+RLlib provides a collection of utilities that abstract away common tasks in RL training.
 
-It is important to understand that the iterators of an execution plan are evaluated *lazily*. This means that no computation happens until the `trainer <#trainers>`__ tries to read the next item from the iterator (i.e., get the next training result for a ``Trainer.train()`` call).
+**Sample Batches** (`sample_batch.py <https://github.com/ray-project/ray/blob/master/rllib/policy/sample_batch.py>`__):
+``SampleBatch`` and `MultiAgentBatch` are the two datatypes that we use for containing timesteps in RLlib. All of our
+RLlib abstractions (policies, replay buffers, etc.) operate using these types.
 
-Execution Plan Concepts
-~~~~~~~~~~~~~~~~~~~~~~~
+**Rollout Workers** (`rollout_worker.py <https://github.com/ray-project/ray/blob/master/rllib/evaluation/rollout_worker.py>`__):
+Rollout workers are an abstraction that wrap a policy (or policies in the case of multi-agent) and an environment.
+From a high-level, we can use rollout workers to collect experiences from the environment by calling
+their ``sample`` function and we can train their policies by calling their ``learn_on_batch`` function.
+By default in RLlib we create a set of workers that can be used for sampling and training. We create a ``WorkerSet``
+object inside of ``setup`` which is called when an RLlib algorithm is created. The ``WorkerSet`` has a ``local_worker``
```

Comment:
Inside of the Trainer's ``setup`` method -> Trainers are explained above, we should reference them here I think.

Reply:
...inside of ``setup``, which..